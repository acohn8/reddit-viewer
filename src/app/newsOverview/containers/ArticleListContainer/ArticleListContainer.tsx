import * as React from 'react';
import { connect } from 'react-redux';

import { Feed, Grid, Divider, Header, Segment } from 'semantic-ui-react';
import ArticlePreview from '../../components/ArticlePreviewComponent/ArticlePreviewComponent';
import { Store } from '../../duck/reducers';

export interface RedditPost {
  subreddit_name_prefixed: string;
  permalink: string;
  url: string;
  created_utc: number;
  description: string;
  thumbnail: string;
  title: string;
  icon: string;
  num_comments: number;
  ups: number;
}

type Iprops = {
  redditPosts: RedditPost[];
};

export const ArticleListContainer = (props: Iprops) => {
  const { redditPosts } = props;
  return (
    <Grid centered columns={2} stackable relaxed="very">
      <Grid.Column style={{ overflowY: 'auto' }}>
        <Header size="large" textAlign="center">
          Trending on Reddit
        </Header>
        <Segment>
          <Feed size="large">
            {redditPosts.map(post => [
              <ArticlePreview
                key={post.permalink}
                link={post.url}
                source={post.subreddit_name_prefixed}
                text={post.description}
                title={post.title}
                image={post.thumbnail}
                postDate={new Date(post.created_utc * 1000)}
                comments={post.num_comments}
                upVotes={post.ups}
                icon="reddit"
              />,
              <Divider />,
            ])}
          </Feed>
        </Segment>
      </Grid.Column>
    </Grid>
  );
};

const mapStateToProps = (state: Store) => ({
  redditPosts: state.article.redditPosts,
});

export default connect(mapStateToProps)(ArticleListContainer);
