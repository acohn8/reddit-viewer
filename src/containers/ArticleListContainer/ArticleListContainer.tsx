import * as React from 'react';
import { connect } from 'react-redux';
import { Divider, Feed, Grid, Header, Segment } from 'semantic-ui-react';

import ArticlePreview from '../../components/ArticlePreviewComponent/ArticlePreviewComponent';
import { StoreState } from '../../types/index';

export interface RedditPost {
  source: string;
  link: string;
  permalink: string;
  postDate: string;
  image: string;
  title: string;
  comments: number;
  upVotes: number;
}

interface Props {
  redditPosts: RedditPost[];
}

export class ArticleListContainer extends React.Component<Props> {
  render() {
    const { redditPosts } = this.props;
    return (
      <Grid centered columns={2} stackable relaxed="very">
        <Grid.Column style={{ overflowY: 'auto' }}>
          <Header size="large" textAlign="center">
            Trending on Reddit
          </Header>
          <Segment>
            <Feed size="large">
              {redditPosts.map(post => [
                <ArticlePreview
                  key={post.link}
                  link={post.link}
                  source={post.source}
                  title={post.title}
                  image={post.image}
                  permalink={post.permalink}
                  postDate={post.postDate}
                  comments={post.comments}
                  upVotes={post.upVotes}
                />,
                <Divider key={`${post.link}div`} />,
              ])}
            </Feed>
          </Segment>
        </Grid.Column>
      </Grid>
    );
  }
}

const mapStateToProps = (state: StoreState) => ({
  redditPosts: state.redditPosts,
});

export default connect(mapStateToProps)(ArticleListContainer);
