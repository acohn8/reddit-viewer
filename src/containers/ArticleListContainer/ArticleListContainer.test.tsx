import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';

import ArticlePreview from '../../components/ArticlePreviewComponent/ArticlePreviewComponent';
import { ArticleListContainer, RedditPost } from './ArticleListContainer';

describe('ArticlePreview', () => {
  const redditPosts: RedditPost[] = [
    {
      comments: 5,
      image: 'image.jpg',
      link: 'www.newyorker.com',
      permalink: 'newyorker.com',
      postDate: 'yesterday',
      source: 'New Yorker',
      title: 'breaking',
      upVotes: 7,
    },
    {
      comments: 5,
      image: 'image.jpg',
      link: 'www.newyorker.com',
      permalink: 'newyorker.com',
      postDate: 'yesterday',
      source: 'New Yorker',
      title: 'breaking',
      upVotes: 7,
    },
  ];
  it('should render correctly', () => {
    const output = shallow(<ArticleListContainer redditPosts={redditPosts} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should render an ArticlePreview for each article in its array of props', () => {
    const output = shallow(<ArticleListContainer redditPosts={redditPosts} />).dive();
    expect(output.find(ArticlePreview).length).toEqual(2);
  });
  it('should not render any ArticlePreview items when given an empty array', () => {
    const output = shallow(<ArticleListContainer redditPosts={[]} />).dive();
    expect(output.find(ArticlePreview).length).toEqual(0);
  });
});
