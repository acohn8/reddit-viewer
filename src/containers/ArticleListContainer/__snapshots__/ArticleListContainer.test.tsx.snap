// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreview should render correctly 1`] = `
<Grid
  centered={true}
  columns={2}
  relaxed="very"
  stackable={true}
>
  <GridColumn
    style={
      Object {
        "overflowY": "auto",
      }
    }
  >
    <Header
      size="large"
      textAlign="center"
    >
      Trending on Reddit
    </Header>
    <Segment>
      <Feed
        size="large"
      >
        <ArticlePreview
          comments={5}
          icon="reddit"
          image="image.jpg"
          key="www.newyorker.com"
          link="www.newyorker.com"
          postDate={1970-01-18T18:47:03.624Z}
          source="New Yorker"
          title="breaking"
          upVotes={7}
        />
        <Divider
          key="www.newyorker.comdiv"
        />
        <ArticlePreview
          comments={5}
          icon="reddit"
          image="image.jpg"
          key="www.newyorker.com"
          link="www.newyorker.com"
          postDate={1970-01-18T18:47:03.624Z}
          source="New Yorker"
          title="breaking"
          upVotes={7}
        />
        <Divider
          key="www.newyorker.comdiv"
        />
      </Feed>
    </Segment>
  </GridColumn>
</Grid>
`;
